packages/sdk/src/conversation/conversation.ts

```diff
@@ -94,18 +95,78 @@ export class ConversationBot {
     if (options.adapter) {
       this.adapter = options.adapter;
     } else {
-      this.adapter = new BotFrameworkAdapter({
-        appId: process.env.BOT_ID,
-        appPassword: process.env.BOT_PASSWORD,
-      });
+      this.adapter = this.createDefaultAdapter(process.env.BOT_ID, process.env.BOT_PASSWORD);
+    }
+
+    if (options.command?.enabled) {
+      this.command = new CommandBot(this.adapter, options.command);
     }
 
-    if (options.command.enabled) {
-      this.command = new CommandBot(this.adapter, options.command.options);
+    if (options.notification?.enabled) {
+      this.notification = new NotificationBot(this.adapter, options.notification);
     }
+  }
+
+  private createDefaultAdapter(appId?: string, appPassword?: string): BotFrameworkAdapter {
+    const adapter = new BotFrameworkAdapter({
+      appId: appId,
+      appPassword: appPassword,
+    });
+
+    // the default error handler
+    adapter.onTurnError = async (context, error) => {
+      // This check writes out errors to console.
+      console.error(`[onTurnError] unhandled error: ${error}`);
+
+      // Send a trace activity, which will be displayed in Bot Framework Emulator
+      await context.sendTraceActivity(
+        "OnTurnError Trace",
+        `${error}`,
+        "https://www.botframework.com/schemas/error",
+        "TurnError"
+      );
+
+      // Send a message to the user
+      await context.sendActivity(`The bot encountered unhandled error: ${error.message}`);
+      await context.sendActivity("To continue to run this bot, please fix the bot source code.");
+    };
+
+    return adapter;
+  }
 
-    if (options.notification.enabled) {
-      this.notification = new NotificationBot(this.adapter, options.notification.options);
+  /**
+   * The request handler to integrate with web request.
+   *
+   * @param req - an Express or Restify style request object.
+   * @param res - an Express or Restify style response object.
+   * @param logic - the additional function to handle bot context.
+   *
+   * @example
+   * For example, to use with Restify:
+   * ``` typescript
+   * // The default/empty behavior
+   * server.post("api/messages", conversationBot.requestHandler);
+   *
+   * // Or, add your own logic
+   * server.post("api/messages", async (req, res) => {
+   *   await conversationBot.requestHandler(req, res, async (context) => {
+   *     // your-own-context-logic
+   *   });
+   * });
+   * ```
+   *
+   * @beta
+   */
+  public async requestHandler(
+    req: WebRequest,
+    res: WebResponse,
+    logic?: (context: TurnContext) => Promise<any>
+  ): Promise<void> {
+    if (logic === undefined) {
```

Comment:
What do you think of providing a default `TeamsActivityHandler` for the `ConversationBot` users? If they want to handler [other event/activity](https://docs.microsoft.com/en-us/microsoftteams/platform/bots/bot-basics?tabs=javascript#bot-logic), they can just implement the default handler's `onXXX` method.

Reply:
IMO, different from `Adapter`, `ConversationBot` has nothing to do with `TeamsActivityHandler`. If providing activity handler, it may confuse developer that *conversation bot requires an activity handler for balabala...*

My suggestion is to add `TeamsActivityHandler` in template or sample code if needed. This `requestHandler` does support to integrate with activity handler:
``` typescript
server.post("api/messages", async (req, res) => {
  await conversationBot.requestHandler(req, res, async (context) => {
    await activityHandler.run(context);
  });
});
```

BTW, providing default `TeamsActivityHandler` may make the SSO scenario complex...